src/test/java/acceptance/queue/QueueSteps.java

```diff
@@ -215,7 +247,10 @@ public void response_queue_contains_expected(List<ResponseRepresentation> expect
         List<String> expectedContents = expectedResponses.stream()
                 .map(responseRepresentation -> responseRepresentation.payload)
                 .collect(Collectors.toList());
-        assertThat("The responses are not correct",responseQueue.getMessageContents(), equalTo(expectedContents));
+
+        List<String> actualContents = queueBasedImplementationRunner.getReceivedMessages();
```

Comment:
We should not change the existing code to support testing.
The assertion should be done on the queue contents.

Reply:
Same reasons as https://github.com/julianghionoiu/tdl-client-java/pull/7#discussion_r222266693

Reply:
Reverted the code but no solution yet on how to recoup